src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11306,6 +11187,131 @@ static bool containsTernaryCollectionToReinterpret(ExpressionSyntax expression)
+                {
+                    // >>=
+                    token1Kind = SyntaxKind.GreaterThanGreaterThanEqualsToken;
+                }
+            }
+
+            if (IsExpectedBinaryOperator(token1Kind))
+                return (token1Kind, SyntaxFacts.GetBinaryExpression(token1Kind));
+
+            if (IsExpectedAssignmentOperator(token1Kind))
+                return (token1Kind, SyntaxFacts.GetAssignmentExpression(token1Kind));
+
+            if (token1Kind == SyntaxKind.SwitchKeyword && this.PeekToken(1).Kind == SyntaxKind.OpenBraceToken)
+                return (token1Kind, SyntaxKind.SwitchExpression);
+
+            if (token1Kind == SyntaxKind.WithKeyword && this.PeekToken(1).Kind == SyntaxKind.OpenBraceToken)
+                return (token1Kind, SyntaxKind.WithExpression);
+
+            // Something that doesn't expand the current expression we're looking at.  Bail out and see if we
+            // can end with a conditional expression.
+            return (SyntaxKind.None, SyntaxKind.None);
+        }
+
+        private SyntaxToken EatOperatorToken(SyntaxKind operatorTokenKind)
+        {
+            // Combine tokens into a single token if needed
+
+            if (operatorTokenKind is SyntaxKind.DotDotToken)
+                return EatDotDotToken();
+
+            if (operatorTokenKind is SyntaxKind.GreaterThanGreaterThanToken or SyntaxKind.GreaterThanGreaterThanEqualsToken)
+            {
+                // >> and >>=
+                // Two tokens need to be consumed here.
+
+                var token1 = EatToken();
+                var token2 = EatToken();
+
+                return SyntaxFactory.Token(
+                    token1.GetLeadingTrivia(),
+                    operatorTokenKind,
+                    token2.GetTrailingTrivia());
+            }
+            else if (operatorTokenKind is SyntaxKind.GreaterThanGreaterThanGreaterThanToken or SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken)
+            {
+                // >>> and >>>=
+                // Three tokens need to be consumed here.
+
+                var token1 = EatToken();
+                _ = EatToken();
+                var token3 = EatToken();
+
+                return SyntaxFactory.Token(
+                    token1.GetLeadingTrivia(),
+                    operatorTokenKind,
+                    token3.GetTrailingTrivia());
+            }
+            else
+            {
+                // Normal operator.  Eat as a single token, converting contextual words cases (like 'with') to a keyword.
+                return this.EatContextualToken(operatorTokenKind);
+            }
+        }
+        
+        private AssignmentExpressionSyntax ParseAssignmentOperator(SyntaxKind operatorExpressionKind, ExpressionSyntax leftOperand, SyntaxToken operatorToken)
+        {
+            Debug.Assert(IsExpectedAssignmentOperator(operatorToken.Kind));
+            Debug.Assert(GetPrecedence(operatorExpressionKind) == Precedence.Assignment);
```

Comment:
Instead of passing `newPrecedence` down thru there, we know it will always be `Precedence.Assignment`, so we just use that.